src/pages/index.astro

```diff
@@ -67,7 +67,7 @@ const cards = [weatherCard, calculatorCard, realEstateCard, forestCard];
         </p>
         <button class="hero--cta">Check offer</button>
       </div>
-      <div id="modal" class="modal-wrapper">
+      <div id="modal" class="modal-wrapper z-10">
```

Comment:
Are you sure this is the best place in the document for a modal?

Reply:
Beside that, I think we should have a separate component for the modal to encapsulate it's logic. Quick refactor might be good in this case